swim/join_sender.go

```diff
@@ -112,6 +130,9 @@ func newJoinSender(node *Node, opts *joinOpts) (*joinSender, error) {
 		node: node,
 	}
 
+	// Parse bootstrap hosts into a map
```

Comment:
This step seems like it would be very easy for a programmer to forget to perform. Parsing bootstrap hosts seems like it should be a detail of the HostProvider, don't you think?